src/hash/rpo/tests.rs

```diff
@@ -51,6 +52,40 @@ fn hash_elements_vs_merge() {
     assert_eq!(m_result, h_result);
 }
 
+#[test]
+fn hash_elements_vs_merge_in_domain() {
```

Comment:
I think the name of this test is a bit confusing: we are not testing whether `hash_elements()` produces a different result from `merge_in_domain()` but rather if `merge_in_domain()` is valid. So, I would probably rename this test to `merge_in_domain()`.

Also, I wonder if we can replace this test with tests which do the following:
- `merge()` and `merge_in_domain()` give the same result when domain is `ZERO`.
- `merge_in_domain()` gives two different results if the domains are different.